### ISSUE TITLE
(When) will there be sqlite support?

### ISSUE THREAD

**Issue by Erkan-Yilmaz** [mediawiki4intranet/Wikilog#7]
see also:

//                https://code.google.com/p/mediawiki-wikilog/issues/detail?id=21
//                 https://code.google.com/p/mediawiki-wikilog/source/detail?r=f6ce06427401d90532cadb7a4d00813dc7e2eb52
//                 https://code.google.com/p/mediawiki-wikilog/source/detail?r=68395d01f8e1292d8eca044f824106b96ca5337a
//                 https://code.google.com/p/mediawiki-wikilog/source/detail?r=837aa060eb7f8400e7512e6a2c1f1b29be352e53

**Comment by vitalif** [comment]
I don't think it should be too hard, you can even try to implement it by yourself :) but...
Do people actually use mediawiki with anything except mysql? I mean, there is a lot of mysql-only extensions, and mediawiki has zero need of an advanced DBMS because it never issues complex queries... :)

**Comment by Erkan-Yilmaz** [comment]
Well, I use MW 1.23.10 now with sqlite 
(I was using MW + mysql on my hoster, but I am self-hosting MW now with sqlite)

**Comment by vitalif** [comment]
Why aren't you self-hosting it with mysql? :-)

**Comment by Erkan-Yilmaz** [comment]
see section "Pros" in https://www.mediawiki.org/wiki/Manual:SQLite#About_SQLite

**Comment by vitalif** [comment]
Ok... I think sqlite isn't really a priority task for me. But pull requests are welcome as always :-)

For internet sites sqlite is not really an option because it's not "not that scalable", it's not scalable at all - only one writer can open the database at a time. And mediawiki is a writer during every request, so it means you can only process requests sequentially with sqlite.

Another big limitation of sqlite is that it does not support ALTER, so you must copy any table to add a column.

**Comment by vitalif** [comment]
I see newer versions are capable of adding columns... but for other manipulations like add index you must still copy the table

**Comment by Erkan-Yilmaz** [comment]
OK, thank you for your time for looking into this.
